src/index.ts

```diff
@@ -283,8 +280,8 @@ function setByPath<
   const parentObject = pathArray.reduce<unknown>((current, pathPart) => {
     if (
       typeof current !== 'object' ||
-      current === null ||
-      !hasOwnProperty.call(current, pathPart)
+      // eslint-disable-next-line no-prototype-builtins
+      !current?.hasOwnProperty(pathPart)
```

Comment:
```suggestion
     current === null ||
      current[pathPart] === undefined || // Trigger dependency tracking for libraries (e.g. vue)
      !hasOwnProperty.call(current, pathPart)
```

The same here. Adding this undefined check should not change behavior, since the check in line 292 would have catched the undefined `parentObject` otherwise.

Reply:
This seems like a good alternative, I see that you already released it, so I will close this